Lib/test/test_winapi.py

```diff
@@ -127,3 +127,37 @@ def test_getshortpathname(self):
 
         # Should contain "PROGRA~" but we can't predict the number
         self.assertIsNotNone(re.match(r".\:\\PROGRA~\d", actual.upper()), actual)
+
+    def test_namedpipe(self):
+        # Deliberately complex name to flush out encoding errors
+        pipe_name = os.urandom(64).replace(b'\0', b'.').decode('latin-1')
```

Comment:
Can it contain `/` and `\`? Trailing space or dot? Try to check manually every possible ending character to ensure that there will be no rare random failures.

Reply:
Good point about trailing dots or spaces since the open uses the "\\\\.\\" prefix in which they get stripped from the end of an opened path. Use "\\\\?\\" to sidestep that issue. The named-pipe filesystem itself allows every Unicode character in pipe names, including control characters, and even slashes since it only implements a root directory. It's not uncommon to create pipe names that contain slashes.

Reply:
If you want to test with a non-ASCII name, you can simply use `TESTFN`, which is across multiple processes and contains a non-ASCII characters supported by the file system. Or use a random ASCII sequence with the `FS_NONASCII` suffix.

Reply:
I tried `TESTFN` and it didn't actually work, but I think that may have been due to getting the prefix wrong. I'll try again.